opal/mca/accelerator/base/accelerator_base_select.c

```diff
@@ -32,116 +32,108 @@
 #include "opal/mca/mca.h"
 #include "opal/mca/accelerator/base/base.h"
 #include "opal/mca/accelerator/accelerator.h"
+#include <string.h>
 
-typedef struct accelerator_opened_component_t {
+typedef struct accelerator_component_t {
   opal_list_item_t super;
   opal_accelerator_base_component_t *accelerator_component;
-  opal_accelerator_base_module_t *accelerator_module;
-} accelerator_opened_component_t;
-
+} accelerator_component_t;
 
 int opal_accelerator_base_select(void)
 {
-
-    int priority = 0, best_priority = -1, ret = 0;
-    opal_list_item_t *item = NULL;
     mca_base_component_list_item_t *cli = NULL;
-    opal_accelerator_base_component_t *component = NULL, *best_component = NULL;
-    opal_accelerator_base_module_t *module = NULL, *best_module = NULL;
-    opal_list_t opened;
-    accelerator_opened_component_t *aoc = NULL;
+    opal_list_item_t *item = NULL;
+    opal_accelerator_base_component_t *component = NULL;
+    opal_accelerator_base_module_t *module = NULL;
+    accelerator_component_t *ac = NULL;
+    opal_list_t ordered_list;
+    bool found_component = false;
 
-    /* Traverse the list of available components; call their init
-       functions. */
+    OBJ_CONSTRUCT(&ordered_list, opal_list_t);
 
-    OBJ_CONSTRUCT(&opened, opal_list_t);
+    /* Traverse the list of available components and create a new ordered list with
+       the NULL Component on the back of the new list */
     OPAL_LIST_FOREACH(cli, &opal_accelerator_base_framework.framework_components, mca_base_component_list_item_t) {
-        component = (opal_accelerator_base_component_t *) cli->cli_component;
+        ac = (accelerator_component_t*) malloc(sizeof(accelerator_component_t));
+        if (NULL == ac) {
+            return OPAL_ERR_OUT_OF_RESOURCE;
+        }
+
+        OBJ_CONSTRUCT(ac, opal_list_item_t);
+        ac->accelerator_component = (opal_accelerator_base_component_t *) cli->cli_component;;
+
+        if (0 == strcmp(cli->cli_component->mca_component_name, "null")) {
+            opal_list_append(&ordered_list, (opal_list_item_t*) ac);
+        }
+        else {
+            opal_list_prepend(&ordered_list, (opal_list_item_t*) ac);
+        }
+    }
+
+    /* Traverse the ordered list of available components;
```

Comment:
I don't think this style of multi-line commenting matches the rest of OMPI code base

Reply:
Updated style to match other multi line comments